File: experiment/switch_ifelse_dict/ifelse.js
function applyTuple(action, num) {
  var result;

  if(action === 'half') {
    result = num / 2;
  } else if(action === 'double') {
    result = num * 2;
  } else if(action === 'triple') {
    result = num * 3;
  } else if(action === 'quadruple') {
    result = num * 4;
  } else { 
    result = num;
  }
  
  return result;
}

const result = applyTuple('double', 2);

File: experiment/switch_ifelse_dict/ifelse.rb
def apply_tuple(action, num)
  if action == :half
    result = num / 2
  elsif action == :double
    result = num * 2
  elsif action == :triple
    result = num * 3
  elsif action == :quadruple
    result = num * 4
  else
    result = num
  end

  result
end

result = apply_tuple :double, 2
